src/components/layout.js

```diff
@@ -26,39 +29,41 @@ const Layout = ({ children, contentClassName }) => {
   `)
 
   return (
-    <div className="flex flex-col min-h-screen antialiased">
-      <Navbar />
-      <main className={`flex-grow ${contentClassName}`}>{children}</main>
-      <footer className="text-gray-800 bg-gray-200 text-center sm:text-left py-3 sm:px-6 sm:flex items-center justify-between">
-        <div className="">
-          <span>
-            ©{data.site.siteMetadata.title}&nbsp;
+    <Search setQuery={setQuery}>
```

Comment:
Can you explain the idea behind wrapping the whole site around the `Search` component?